internal/controllers/clusterextension_controller.go

```diff
@@ -470,16 +472,31 @@ func (d *DefaultInstalledBundleGetter) GetInstalledBundle(ctx context.Context, e
 		return nil, err
 	}
 
-	release, err := cl.Get(ext.GetName())
+	rel, err := cl.Get(ext.GetName())
 	if err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
 		return nil, err
 	}
-	if release == nil {
+	if rel == nil {
 		return nil, nil
 	}
 
+	switch {
+	case rel.Info.Status == release.StatusUnknown:
+		return nil, fmt.Errorf("installation status is unknown")
+	case rel.Info.Status == release.StatusDeployed:
+	case rel.Info.Status == release.StatusUninstalled:
+	case rel.Info.Status == release.StatusSuperseded:
+	case rel.Info.Status == release.StatusFailed:
+	case rel.Info.Status == release.StatusUninstalling:
+		return nil, fmt.Errorf("installation is still pending: %s", rel.Info.Status)
+	case rel.Info.Status.IsPending():
+		return nil, fmt.Errorf("installation is still pending: %s", rel.Info.Status)
+	default:
+		return nil, fmt.Errorf("unknown installation status: %s", rel.Info.Status)
+	}
```

Comment:
Nit: combining cases might make this a little more terse. Also it might make sense to not even enter the switch if the status is still pending.
```suggestion
        if rel.Info.Status.IsPending() {
                return nil, fmt.Errorf("installation is still pending: %s", rel.Info.Status)
        }

	switch rel.Info.Status {
	case release.StatusUnknown:
		return nil, fmt.Errorf("installation status is unknown")
	case release.StatusDeployed, release.StatusUninstalled, release.StatusSuperseded, release.StatusFailed:
	case rel.Info.Status == release.StatusUninstalling:
		return nil, fmt.Errorf("installation is still pending: %s", rel.Info.Status)
	default:
		return nil, fmt.Errorf("unknown installation status: %s", rel.Info.Status)
	}
```

Reply:
Not a fan of the longer lines. Splitting out the `IsPending()` makes it seem as though it's separate when it's not; it's just checking the set of `StatusPendingXxxxx` values.
Also, this happens to match the order they are defined by Helm.